src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java

```diff
@@ -165,21 +165,57 @@ protected Object decode(
                         if (BitUtil.check(alarm, 2)) {
                             position.addAlarm(Position.ALARM_FALL_DOWN);
                         }
+                        if (BitUtil.check(alarm, 4)) {
+                            position.addAlarm(BitUtil.check(alarm, 26) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 1);
+                        }
+                        if (BitUtil.check(alarm, 5)) {
+                            position.addAlarm(BitUtil.check(alarm, 27) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 2);
+                        }
+                        if (BitUtil.check(alarm, 6)) {
+                            position.addAlarm(BitUtil.check(alarm, 28) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 3);
+                        }
+                        if (BitUtil.check(alarm, 7)) {
+                            position.addAlarm(BitUtil.check(alarm, 29) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 4);
+                        }
                         if (BitUtil.check(alarm, 8)) {
                             position.addAlarm(Position.ALARM_POWER_OFF);
                         }
                         if (BitUtil.check(alarm, 9)) {
                             position.addAlarm(Position.ALARM_POWER_ON);
                         }
+                        if (BitUtil.check(alarm, 10)) {
+                            position.addAlarm(Position.ALARM_MOVEMENT);
+                            position.set(Position.KEY_MOTION, true);
```

Comment:
You shouldn't use both here.

Reply:
I think Bit 10 and 11 are actual Motion / No Motion alarms, and do not indicate whether the device is moving. 
You can configure the time after which the alarms trigger.